### PR TITLE
Add vanilla CSS support to Select

### PR DESCRIPTION
**Pull request by benoitgrelard**

### CODE REVIEW

File: components/demos/Select/css/index.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import * as Select from '@radix-ui/react-select';
+import classnames from 'classnames';
+import { CheckIcon, ChevronDownIcon, ChevronUpIcon } from '@radix-ui/react-icons';
+import './styles.css';
+
+export const SelectDemo = () => (
```
Comment (andy-hook) [inline components/demos/Select/css/index.jsx:7]:
```suggestion
const SelectDemo = () => (
```

File: components/demos/Select/css/index.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+);
+
+const SelectContent = React.forwardRef(({ children, className, ...props }, forwardedRef) => {
+  return (
+    <Select.Portal>
+      <Select.Content
+        {...props}
+        ref={forwardedRef}
+        className={classnames('SelectContent', className)}
+      >
```
Comment (andy-hook) [inline components/demos/Select/css/index.jsx:66]:
```suggestion
      <Select.Content
        className={classnames('SelectContent', className)}
        {...props}
        ref={forwardedRef}
      >
```

I think we decided to always place `className` first

File: components/demos/Select/css/index.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        {children}
+      </Select.Content>
+    </Select.Portal>
+  );
+});
+
+const SelectItem = React.forwardRef(({ children, className, ...props }, forwardedRef) => {
+  return (
+    <Select.Item {...props} ref={forwardedRef} className={classnames('SelectItem', className)}>
```
Comment (andy-hook) [inline components/demos/Select/css/index.jsx:75]:
```suggestion
    <Select.Item className={classnames('SelectItem', className)} {...props} ref={forwardedRef}>
```

File: components/demos/Select/css/index.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      <Select.ItemText>{children}</Select.ItemText>
+      <Select.ItemIndicator>
```
Comment (andy-hook) [inline components/demos/Select/css/index.jsx:77]:
```suggestion
      <Select.ItemIndicator className="SelectItemIndicator">
```

I haven't tested if this is what's missing but the indicator is currently on the wrong side:

CleanShot 2022-11-01 at 10 39 58@2x

File: data/primitives/components/select/1.1.1.mdx [hunk 5]
```diff
@@ -542,21 +540,19 @@ Used to visually separate items in the select.
...
-});
-
 export default () => (
   <Select.Root>
     <Select.Trigger>…</Select.Trigger>
     <Select.Portal>
       <Select.Content>
         <Select.Viewport>
-          <StyledItem __disabled__>…</StyledItem>
+          <StyledItem __className__="SelectItem" __disabled__>
```
Comment (andy-hook) [inline data/primitives/components/select/1.1.1.mdx:553]: These should be `Select.Item`

File: data/primitives/components/select/1.1.1.mdx [hunk 6]
```diff
@@ -566,21 +562,24 @@ export default () => (
...
 import * as Select from '@radix-ui/react-select';
 
-const StyledTrigger = styled(Select.Trigger, {
-  '&[data-placeholder]': { color: 'gainsboro' },
-});
-
 export default () => (
   <Select.Root>
-    <SelectTrigger>
+    <SelectTrigger __className__="SelectTrigger">
```
Comment (andy-hook) [inline data/primitives/components/select/1.1.1.mdx:582]: `Select.Trigger`

File: data/primitives/components/select/1.1.1.mdx [hunk 7]
```diff
@@ -656,7 +656,7 @@ export default () => (
         <Select.Viewport>
           <Select.Item>
             <Select.ItemText>
-              <Image src="…" />
+              <img src="…" />
```
Comment (andy-hook) [inline data/primitives/components/select/1.1.1.mdx:659]: I just updated a [similar example](https://github.com/radix-ui/website/pull/520/files#diff-8ebb301faef9380a4588b049166c4364654661942ff0c9eb0fad87aee47f8d1aR1232) and swapped it out to use icons as that seemed a more common use case. Let me know if you'd prefer it to match this or vice versa though.

File: components/demos/Select/stitches/index.jsx [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+      <SelectViewport>
+        <Select.Group>
+          <SelectLabel>Fruits</SelectLabel>
+          <SelectItem value="apple">Apple</SelectItem>
+          <SelectItem value="banana">Banana</SelectItem>
+          <SelectItem value="blueberry">Blueberry</SelectItem>
+          <SelectItem value="grapes">Grapes</SelectItem>
+          <SelectItem value="pineapple">Pineapple</SelectItem>
+        </Select.Group>
+
```
Comment (andy-hook) [inline components/demos/Select/stitches/index.jsx:28]: Whitespace helps a lot with these longer examples, perhaps I should do that for ContextMenu.

File: components/demos/Select/css/index.jsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        </Select.Group>
+      </Select.Viewport>
+      <Select.ScrollDownButton className="SelectScrollButton">
+        <ChevronDownIcon />
+      </Select.ScrollDownButton>
+    </SelectContent>
+  </Select.Root>
+);
+
+const SelectContent = React.forwardRef(({ children, className, ...props }, forwardedRef) => {
```
Comment (andy-hook) [inline components/demos/Select/css/index.jsx:59]: I went the other way with ContextMenu and removed the abstractions, do you think it'd be better to keep it as you have here? I could go either way tbh.

File: data/primitives/components/select/1.1.1.mdx (hunk 7 above)
Comment (benoitgrelard) [reply data/primitives/components/select/1.1.1.mdx:659]: I think I prefer img for the simplicity

File: components/demos/Select/stitches/index.jsx (hunk 8 above)
Comment (benoitgrelard) [reply components/demos/Select/stitches/index.jsx:28]: With the groups it helps yeah.

File: components/demos/Select/css/index.jsx (hunk 9 above)
Comment (andy-hook) [reply components/demos/Select/css/index.jsx:59]: > I think maybe we don't abstract what's not repeated (so for example, I should remove the portal/content abstraction) but for repeated items we do?

This sounds good.

**Comment by benoitgrelard** [comment]
All done 👍

**Review by andy-hook — APPROVED** [review]
Looks great 💯

**Pull request merged**